test/udp_backend_test.rb

```diff
@@ -132,4 +132,16 @@ def test_io_error_should_not_raise_but_log
     @logger.expects(:error)
     StatsD.increment('fail')
   end
+
+  def test_thread_error_does_not_raise
+    pid = fork do
+      Signal.trap('TERM') do
+        StatsD.increment('exiting')
+      end
+
+      sleep 100
+    end
+
+    Process.kill('TERM', pid)
```

Comment:
The thing is that it might, or might not be sent. I don't think we have any guarantees. Let me 👀